v-next/core/test/internal/tasks/task-manager.ts

```diff
@@ -875,4 +877,661 @@ describe("TaskManagerImplementation", () => {
+                        param2: "param2Value",
+                        flag2: true,
+                      },
+                    );
+                  },
+                )
+                .build(),
+            ],
+          },
+        ],
+      });
+
+      const task1 = hre.tasks.getTask("task1");
+      await task1.run({
+        param1: "param1Value",
+        flag1: true,
+        posParam: "posValue",
+        varParam: ["varValue1", "varValue2"],
+        param2: "param2Value",
+        flag2: true,
+      });
+    });
+
+    it("should run a task with arguments and resolve their default values", async () => {
+      const hre = await createHardhatRuntimeEnvironment({
+        plugins: [
+          {
+            id: "plugin1",
+            tasks: [
+              new NewTaskDefinitionBuilderImplementation("task1")
+                .addNamedParameter({
+                  name: "param1",
+                  defaultValue: "param1DefaultValue",
+                })
+                .addFlag({ name: "flag1" })
+                .addPositionalParameter({
+                  name: "posParam",
+                  defaultValue: "posValue",
+                })
+                .addVariadicParameter({
+                  name: "varParam",
+                  defaultValue: ["varValue1", "varValue2"],
+                })
+                .setAction((args) => {
+                  assert.deepEqual(args, {
+                    param1: "param1DefaultValue",
+                    flag1: false,
+                    posParam: "posValue",
+                    varParam: ["varValue1", "varValue2"],
+                  });
+                })
+                .build(),
+            ],
+          },
+        ],
+      });
+
+      const task1 = hre.tasks.getTask("task1");
+      await task1.run({});
+    });
+
+    it("should run an empty task that was overriden", async () => {
+      let overrideTaskRun = false;
+      const hre = await createHardhatRuntimeEnvironment({
+        plugins: [
+          {
+            id: "plugin1",
+            tasks: [
+              {
+                id: ["task1"],
+                description: "description1",
+                type: TaskDefinitionType.EMPTY_TASK,
+              },
+              new TaskOverrideDefinitionBuilderImplementation("task1")
+                .setAction(async (args, _hre, runSuper) => {
+                  await runSuper(args);
+                  overrideTaskRun = true;
+                })
+                .build(),
+            ],
+          },
+        ],
+      });
+
+      const task1 = hre.tasks.getTask("task1");
+      assert.equal(overrideTaskRun, false);
+      await task1.run({});
+      assert.equal(overrideTaskRun, true);
+    });
+
+    it("should run a task with an action url", async () => {
+      const actionUrl = pathToFileURL(
+        path.resolve(
+          import.meta.dirname,
+          "./fixture-projects/file-actions/action-fn.js",
+        ),
+      ).toString();
```

Comment:
FYI:

`import.meta.resolve("./fixture-projects/file-actions/action-fn.js")` is equivalent

Reply:
Nice, I'll change it just to have it as a reference for the future.